services/catalog/src/simcore_service_catalog/db/repositories/dags.py

```diff
@@ -13,48 +13,55 @@
 class DAGsRepository(BaseRepository):
     async def list_dags(self) -> List[DAGAtDB]:
         dagraphs = []
-        async for row in self.connection.execute(dags.select()):
-            if row:
-                dagraphs.append(DAGAtDB(**row))
+        async with self.db_engine.acquire() as conn:
+            async for row in conn.execute(dags.select()):
+                if row:
+                    dagraphs.append(DAGAtDB(**row))
         return dagraphs
 
     async def get_dag(self, dag_id: int) -> Optional[DAGAtDB]:
-        stmt = dags.select().where(dags.c.id == dag_id)
-        row: RowProxy = await (await self.connection.execute(stmt)).first()
-        if row:
-            return DAGAtDB(**row)
-        return None
+        async with self.db_engine.acquire() as conn:
+            row: RowProxy = await (
+                await conn.execute(dags.select().where(dags.c.id == dag_id))
+            ).first()
+            if row:
+                return DAGAtDB(**row)
```

Comment:
could you reduce indentation by 1 level here, to guaranty an early release of the connection?

Reply:
actually learned something this morning about scope of with, if and for in python... very weird though. thanks

Reply:
but now. what if row is None.. this will fail right?

Reply:
ah no. I keep the whole block all good. thanks